src/wallet/wallet.cpp

```diff
@@ -3324,10 +3325,10 @@ void CWallet::GetAllReserveKeys(std::set<CKeyID>& setAddress) const
     CWalletDB walletdb(*dbw);
 
     LOCK2(cs_main, cs_wallet);
-    BOOST_FOREACH(const int64_t& id, setKeyPool)
+    for (const KeypoolCache &poolPair : setKeyPool)
     {
         CKeyPool keypool;
-        if (!walletdb.ReadPool(id, keypool))
+        if (!walletdb.ReadPool(poolPair.nIndex, keypool))
```

Comment:
I am wondering if it is not better to just change `ReadPool` signature to take KeyPoolCache instead of index.